cfaster/scrapers/utils/urls.py

```diff
@@ -3,15 +3,48 @@
 
 logger = logging.getLogger(__name__)
 
+def url_filter(string, to_remove):
+    if string.startswith(to_remove):
+        return string[len(to_remove):]
+    else:
+        return string
+
 
 def validate_url(url, expected_domain):
     logger.info('Validating url {} with expected domain {}'.format(
         url, expected_domain))
     url = url.strip()
-    reg = r'^(?:https?://)?(?:www.)?(\w+)(?:.\w+)+(?:/\w*)*$'
-    match = re.search(reg, url)
-    if match is None:
+
+    if expected_domain not in url:
         return False
-    if len(match.groups()) and match.group(1) == expected_domain:
-        return True
-    return False
+
+    for leader in ('https://', 'http://', 'www', 'ftp://'):
```

Comment:
I think `www.` would be the right choice instead of `www`. It is not giving us any errors because the split on line 28 fixes it, but it would be interesting not to propagate any error to be fixed in another line.